assets/helpers/details.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+const isDetailsSupported: boolean = ('open' in document.createElement('details'));
+
+function toggleDetailsState(details: HTMLDetailsElement) {
+  if (details.hasAttribute('open')) {
+    details.removeAttribute('open');
+  } else {
+    details.setAttribute('open', 'open');
+  }
+}
+
+function handleEvent(event: UIEvent) {
+  let targetNode: Element = (event.target: any);
+
+  while (targetNode.nodeName.toLowerCase() !== 'summary' && targetNode !== document && targetNode.parentElement) {
+    targetNode = targetNode.parentElement;
+  }
+  if (targetNode.nodeName.toLowerCase() === 'summary' && targetNode.parentElement instanceof HTMLDetailsElement) {
```

Comment:
is there any chance that  `targetNode.nodeName` will return `undefined`?

Reply:
Nope.